src/MJanssen/Provider/RoutingServiceProvider.php

```diff
@@ -49,97 +50,121 @@ public function boot(Application $app)
+    {
+        $availableMethods = array('get', 'put', 'post', 'delete', 'options', 'head');
+        foreach(array_map('strtolower', $methods) as $method){
+            if(!in_array($method, $availableMethods)){
+                throw new InvalidArgumentException('Method "' . $method . '" is not valid, only the following methods are allowed: get, put, post, delete, options, head');
+            }
         }
     }
 
     /**
+     * Validates the given $route Array
+     *
      * @param $route
      * @throws \InvalidArgumentException
      */
-    protected function validateRoute($route)
+    protected function validateRoute(&$route)
     {
-        if(!isset($route['pattern']) || !isset($route['method']) || !isset($route['controller'])) {
+        if (!isset($route['pattern']) || !isset($route['method']) || !isset($route['controller'])) {
             throw new InvalidArgumentException('Required parameter (pattern/method/controller) is not set.');
         }
 
         $arrayParameters = array('method', 'assert', 'value');
 
-        foreach($arrayParameters as $parameter) {
-            if(isset($route[$parameter]) && !is_array($route[$parameter])) {
+        foreach ($arrayParameters as $parameter) {
+            if (isset($route[$parameter]) && !is_array($route[$parameter])) {
                 throw new InvalidArgumentException(sprintf(
                     '%s is not of type Array (%s)',
                     $parameter, gettype($route[$parameter])
                 ));
             }
         }
-    }
-
 
-    /**
-     * @param Application $app
-     * @param $route
-     * @throws \InvalidArgumentException
-     */
-    protected function addRouteByMethod(Application $app, $route)
-    {
-        $controller = $this->getController($app, $route);
-
-        $supportedProperties = array('value', 'assert', 'before', 'after');
-
-        foreach($supportedProperties AS $property) {
-            if(isset($route[$property])) {
-                $this->addActions($controller, $route[$property], $property);
-            }
-        }
+        $this->validateMethods($route['method']);
 
-        if(isset($route['scheme'])) {
-            if('https' === $route['scheme']) {
-                $controller->requireHttps();
-            }
-        }
+        $this->sanitizeRouteName($route['routeName']);
     }
 
+
     /**
-     * @param Application $app
-     * @param $route
-     * @return mixed|Controller
-     * @throws \InvalidArgumentException
+     * Sanitizes the routeName for named route:
+     *
+     * - replaces '/', ':', '|', '-' with '_'
+     * - removes special characters
+     *
+     *  Algorithm copied from \Silex\Controller->generateRouteName
+     *  see: https://github.com/silexphp/Silex/blob/1.2/src/Silex/Controller.php
+     *
+     * @param string $routeName
+     * @return string
      */
-    protected function getController(Application $app, $route)
+    protected function sanitizeRouteName(&$routeName)
```

Comment:
If it's passed by reference why return it? Could you please just remove the pass by reference and return eouteName by default?